### PR TITLE
airbyte-lib: Generate docs

### PR DESCRIPTION
**Pull request by flash1293**
This PR generates documentation for the public interface of airbyte-lib. It will be integrated into our documentation system in a separate PR.

* Use a python script using pdoc to generate the documentation (without any chrome so it can easily be embedded)
* Add a unit test that checks whether re-running the doc generation will generate a git diff (to enforce committing docs changes)

### CODE REVIEW

File: airbyte-lib/.gitattributes
Comment (aaronsteers) [inline airbyte-lib/.gitattributes:1]: 💙

File: airbyte-lib/tests/docs_tests/test_docs_checked_in.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import docs
+
+
+def test_docs_checked_in():
+    """
+    Docs need to be generated via `poetry run generate-docs` and checked in to the repo.
+
+    This test runs the docs generation and compares the output with the checked in docs.
+    It will fail if there are any differences.
```
Comment (aaronsteers) [inline airbyte-lib/tests/docs_tests/test_docs_checked_in.py:13]: Nice. 👍

**Review by aaronsteers — APPROVED** [review]
Looks good! Any other action items to call out?

I noticed Vercel both appears to skip this PR as not relevant. Not sure if we should (eventually) have integration with Vercel - or else have another way to preview airbyte-lib docs. Wdyt?

**Comment by flash1293** [comment]
> I noticed Vercel both appears to skip this PR as not relevant. Not sure if we should (eventually) have integration with Vercel - or else have another way to preview airbyte-lib docs. Wdyt?

This is configurable on the vercel side - after we actually merge the PR that's integrating the airbyte-lib docs in our doc system, I will adjust this config to also re-build if those generated docs change.

Does that sound good? Alternatively we could create that integration now with a hidden page in our docs, but I would like to keep it clean if possible, to not accidentally render the wrong things

**Pull request merged**